src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -597,21 +858,37 @@ else if (!hostIdsInCurrentRing.contains(nodeHistory.hostId))
                 if (priorityHostId != null && !myId.equals(priorityHostId))
                 {
                     logger.info("Priority list is not empty and I'm not the first node in the list, not my turn." +
-                                "First node in priority list is {}", ClusterMetadata.current().directory.addresses.get(NodeId.fromUUID(priorityHostId)));
+                                "First node in priority list is {}", getBroadcastAddress(priorityHostId));
                     return NOT_MY_TURN;
                 }
+
                 if (myId.equals(priorityHostId))
                 {
                     //I have a priority for repair hence its my turn now
                     return MY_TURN_DUE_TO_PRIORITY;
                 }
-                // get the longest unrepaired node from the nodes which are not running repair
-                AutoRepairHistory defaultNodeToBeRepaired = getHostWithLongestUnrepairTime(currentRepairStatus.historiesWithoutOnGoingRepair);
-                //check who is next, which is helpful for debugging
-                logger.info("Next node to be repaired for repair type {} by default: {}", repairType, defaultNodeToBeRepaired);
-                if (defaultNodeToBeRepaired != null && defaultNodeToBeRepaired.hostId.equals(myId))
+
+                // Determine if this node is the most eligible host to repair.
+                AutoRepairHistory nodeToBeRepaired = getMostEligibleHostToRepair(repairType, currentRepairStatus, myId);
+                if (nodeToBeRepaired != null)
+                {
+                    if (nodeToBeRepaired.hostId.equals(myId))
+                    {
+                        logger.info("This node is selected to be repaired for repair type {}", repairType);
+                        return MY_TURN;
+                    }
+
+                    // log which node is next, which is helpful for debugging
+                    logger.debug("Next node to be repaired for repair type {}: {} ({})", repairType,
```

Comment:
Can we keep this _info_? Often useful debugging who should be the next. It was already _info_ at line no 611

Reply:
sounds good, one thing I should note here is that we currently short circuit evaluating all nodes at line 600 (`takeWhile`), so its possible we will not log this if we encounter the current nodes repair history and see that it isn't eligible for repair, in this case we wouldn't log at all other than the log line saying we are deferring repair because replica is found.   I think that's probably ok though as we still get a justification for why the repair isn't happening in either case (1. there's a more eligible node or 2. this node has replica being repaired).